js/okcoin.js

```diff
@@ -2575,6 +2586,93 @@ module.exports = class okcoin extends Exchange {
         return address;
     }
 
+    async transfer (code, amount, fromAccount, toAccount, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const accountsByType = this.safeValue (this.options, 'accountsByType', {});
+        const fromId = this.safeString (accountsByType, fromAccount, fromAccount);
+        const toId = this.safeString (accountsByType, toAccount, toAccount);
+        const request = {
+            'amount': this.currencyToPrecision (code, amount),
+            'currency': currency['id'],
+            'from': fromId,
+            'to': toId,
+            'type': '0',
+        };
+        if (fromAccount === 'main') {
```

Comment:
This `'main'` string isn't listed in `options['accountsByType']`.
Also, here, after the safeString replacements on lines 2593-2594, we should use the `fromId/toId` instead of `fromAccount/toAccount` for conditionals.

Reply:
For this exchange it can be transfers between account types, and also between the `main` account and subaccounts. `main` is used to differentiate between one type of transfer and the other